lib/internals/root.js

```diff
@@ -27,4 +35,89 @@ module.exports = {
       credentials
     }
   }
+
+  /**
+   * Function that will execute a function and pass the response to a callback to determine whether to retry the execution
+   *
+   * @param {Function} toExecute Function to execute
+   * @param {Function} onResponse Callback to be called that will determine whether to retry the execution or not
+   * @returns {Promise<*>} The outcome of toExecute
+   */
+  const retryable = async (toExecute, onResponse) => {
+    let iterationCount = 1
+    let shouldRetry = true
+    let err
+    let response
+
+    while (shouldRetry) {
+      try {
+        response = await toExecute()
+      } catch (e) {
+        err = e
+      }
+
+      /**
+       * The callback to be executed if the request is to be retried. If delayMs is provided, that amount of milliseconds will be waited before retrying
+       *
+       * @param {Number} delayMs
+       * @returns {Promise<void>}
+       */
+      const retry = async (delayMs) => {
+        debug(`Retrying after ${iterationCount} try after a delay of ${delayMs || 0}ms`)
+
+        iterationCount++
+
+        if (typeof delayMs === 'number') {
```

Comment:
Consider if this "if" statement is required given that delayMs should always be a number